specs/Workers.md

```diff
@@ -0,0 +1,1402 @@
+                            CHECK_FAILURE(args->get_ActiveServiceWorker(&serviceWorker));
+                        
+                            std::wstringstream message;
+
+                             // Subscribe to worker destroyed event
+                             CHECK_FAILURE(serviceWorker->add_Destroyed(
+                                    Callback<
+                                        ICoreWebView2ServiceWorkerDestroyedEventHandler>(
+                                        [this](
+                                            ICoreWebView2ServiceWorker* sender,
+                                            IUnknown* args) -> HRESULT
+                                        {
+                                            /*Cleanup on worker destruction*/
+                                            return S_OK;
+                                        })
+                                        .Get(),
+                                    nullptr));
+
+                                // Send a list of resources to the service worker to cache.
+                                PostMessageToWorker(serviceWorker);
+
+                            return S_OK;
+                        })
+                        .Get(),
+                nullptr));
+                
+                    LogMessage(L"Service worker has been registered with a scope:  " + std::wstring(scopeUri.get()));
+                  }
+
+                  return S_OK;
+              })
+              .Get(),
+          &m_serviceWorkerRegisteredToken));
+    }
+```
+
+## Retrieving Service Worker Registrations.
+The following example below illustrates how to retrieve and present details
+about all service worker registrations linked to the WebView2 profile. This
+yields a list of `CoreWebView2ServiceWorkerRegistration` objects, enabling the
+host application to communicate directly with the service workers via post
+messaging.
+
+## .NET/WinRT
+```c#
+    private async void GetServiceWorkerRegistrationsExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        try
+        {
+            if (ServiceWorkerManager_ != null)
+            {
+                ServiceWorkerManager_ = WebViewProfile.ServiceWorkerManager;
+            }
+            IReadOnlyList<CoreWebView2ServiceWorkerRegistration> registrationList = await 
+            ServiceWorkerManager_.GetServiceWorkerRegistrationsAsync();
+
+            int registrationCount = registrationList.Count;
+            StringBuilder messageBuilder = new StringBuilder();
+            messageBuilder.AppendLine($"No of service workers registered: {registrationCount}");
+
+            for (int i = 0; i < registrationList.Count(); ++i)
+            {
+                var scopeUri = registrationList[i].ScopeUri;
+
+                messageBuilder.AppendLine($"Scope: {scopeUri}");
+            };
+
+            LogMessage(messageBuilder.ToString());
+        }
+        catch (NotImplementedException exception)
```

Comment:
Why is this necessary? Apps should never need to have catch blocks in their code. (Exceptions should only happen on app bugs.)

Reply:
See above.